### DISCUSSION TITLE
What to use instaed of useSigner in the version   "wagmi": "^2.8.3",

### DISCUSSION THREAD

**Discussion by Lakszy** [wevm/wagmi#4297] [Q&A] [answered]
const collToken = new ethers.Contract(
"ERC20 token address"
erc20Abi,
 signer
 );
 
 
 How to get this signer

**Comment by Lakszy** [comment]
This is working earlier in the previous versions 
`const { data: Signer, isError, isLoading } = useSigner();`

This too (not working now):
` const provider = new providers.Web3Provider(transport, network)`
`const signer = provider.getSigner(account.address)`

**Reply by tmm** [reply]
Check out the docs https://wagmi.sh/react/guides/ethers

**Reply by Lakszy** [reply, marked as answer]
> Check out the docs https://wagmi.sh/react/guides/ethers

Thanks 
But please could be more specific 

import { providers } from 'ethers'
import { useMemo } from 'react'
import type { Account, Chain, Client, Transport } from 'viem'
import { Config, useConnectorClient } from 'wagmi'

export function clientToSigner(client: Client<Transport, Chain, Account>) {
  const { account, chain, transport } = client
  const network = {
    chainId: chain.id,
    name: chain.name,
    ensAddress: chain.contracts?.ensRegistry?.address,
  }
  const provider = new providers.Web3Provider(transport, network)
  const signer = provider.getSigner(account.address)
  return signer
}

/** Hook to convert a Viem Client to an ethers.js Signer. */
export function useEthersSigner({ chainId }: { chainId?: number } = {}) {
  const { data: client } = useConnectorClient<Config>({ chainId })
  return useMemo(() => (client ? clientToSigner(client) : undefined), [client])
}
